Userland/Libraries/LibWeb/PixelUnits.h

```diff
@@ -278,6 +279,85 @@ constexpr CSSPixels operator*(unsigned long left, CSSPixels right) { return righ
 inline float operator*(float left, CSSPixels right) { return right.to_float() * left; }
 inline double operator*(double left, CSSPixels right) { return right.to_double() * left; }
 
+class CSSPixelFraction {
```

Comment:
cool, I think it makes to represent temporary evaluation results using fractions to have maximum precision.

although the only thing that doesn't seem right for me is that `CSSPixelFraction` could be used outside of expression. could we make `CSSPixelFraction` to be `CSSPixels` implementation detail that should never leak outside of it?

Reply:
I'm not sure how that would work, especially given the usage in the second commit in this PR, where we take the `min()` of multiple fractions. We need to be able to store a fraction as a variable for that to work with any reasonable amount of readability, I think.

Reply:
It's also worth noting that with it as a public class, we should potentially be able to expand it to cover more use cases, like retaining more precision when multiplying or dividing fractions. I haven't seen a use for that yet, but it's certainly an option if any use case arises.